$MirOS: ports/graphics/cairo/patches/patch-configure_in,v 1.2 2005/09/22 13:55:57 bsiegert Exp $
--- configure.in.orig	Wed Aug 24 18:25:48 2005
+++ configure.in	Fri Oct  7 23:38:07 2005
@@ -89,18 +89,21 @@ AC_ARG_ENABLE(xlib,
   [  --disable-xlib          Disable cairo's Xlib backend],
   [use_xlib=$enableval], [use_xlib=yes])
 
+XRENDER_LIBS=
 if test "x$use_xlib" = "xyes"; then
   dnl Check for Xrender header files if the Xrender package is not installed:
   PKG_CHECK_MODULES(XRENDER, xrender >= 0.6, [
     XRENDER_REQUIRES=xrender
     use_xlib=yes], [
-      AC_CHECK_HEADER(X11/extensions/Xrender.h, [
       AC_PATH_XTRA
+      CPPFLAGS="$CPPFLAGS $X_CFLAGS"
+      AC_CHECK_HEADER(X11/extensions/Xrender.h, [
       XRENDER_LIBS="$X_LIBS -lXrender -lXext -lX11 $X_EXTRA_LIBS"
       use_xlib=yes], [
 		use_xlib="no (requires Xrender http://freedesktop.org/Software/xlibs)"])])
   _CHECK_FUNCS_WITH_FLAGS(XrmFinalize, $XRENDER_CFLAGS, $XRENDER_LIBS)
 fi
+AC_SUBST(XRENDER_LIBS)
 
 AM_CONDITIONAL(CAIRO_HAS_XLIB_SURFACE, test "x$use_xlib" = "xyes")
 if test "x$use_xlib" = "xyes"; then
@@ -197,7 +200,7 @@ AC_ARG_ENABLE(png,
 if test "x$use_png" = "xyes"; then
   use_png=no
   # libpng13 is GnuWin32's libpng-1.2.8 :-(
-  for l in libpng12 libpng13 libpng10 ; do
+  for l in libpng libpng12 libpng13 libpng10 ; do
     if $PKG_CONFIG --exists $l ; then
       PNG_REQUIRES=$l
       use_png=yes
@@ -209,7 +212,16 @@ if test "x$use_png" = "xyes"; then
     # Sets PNG_CFLAGS, PNG_LIBS
     PKG_CHECK_MODULES(PNG, $PNG_REQUIRES)
   else
-    AC_MSG_WARN([Could not find libpng in the pkg-config search path])
+    AC_CHECK_LIB(png, png_read_info,
+      [AC_CHECK_HEADER(libpng/png.h,
+	[use_png=yes; PNG_CFLAGS='-I${PREFIX}/include/libpng'],
+	[AC_CHECK_HEADER(png.h,
+	  [use_png=yes; PNG_CFLAGS=],
+	  use_png=no)])],
+      AC_MSG_WARN(PNG library not found), -lz -lm)
+    if test "x$use_png" = xyes; then
+      PNG_LIBS='-lpng -lz -lm'
+    fi
   fi    
 fi
 
@@ -466,10 +478,14 @@ GTK_DOC_CHECK([1.3])
 dnl ===========================================================================
 dnl Check for recent pkg-config which supports Requires.private
 
+changequote(,)dnl
+
 case `$PKG_CONFIG --version` in
-0.?|0.1[0-7]) PKGCONFIG_REQUIRES="Requires"; ;;
+0.?|0.1[0-7]*) PKGCONFIG_REQUIRES="Requires"; ;;
 *) PKGCONFIG_REQUIRES="Requires.private"; ;;
 esac
+
+changequote([,])dnl
 
 AC_SUBST(PKGCONFIG_REQUIRES)
 
